src/metpy/plots/station_plot.py

```diff
@@ -304,6 +295,75 @@ def plot_barb(self, u, v, **kwargs):
         else:
             self.barbs = self.ax.barbs(self.x, self.y, u, v, **defaults)
 
+    def plot_arrow(self, u, v, **kwargs):
+        r"""At the center of the station model plot wind arrows.
+
+        Additional keyword arguments given will be passed onto matplotlib's
+        :meth:`~matplotlib.axes.Axes.quiver` function; this is useful for specifying things
+        like color or line width.
+
+        Parameters
+        ----------
+        u : array-like
+            The data to use for the u-component of the arrows.
+        v : array-like
+            The data to use for the v-component of the arrows.
+        plot_units: `pint.unit`
+            Units to plot in (performing conversion if necessary). Defaults to given units.
+        kwargs
+            Additional keyword arguments to pass to matplotlib's
+            :meth:`~matplotlib.axes.Axes.barbs` function.
+
+        See Also
+        --------
+        plot_barb, plot_parameter, plot_symbol, plot_text
+
+        """
+        kwargs = self._make_kwargs(kwargs)
+
+        # If plot_units specified, convert the data to those units
+        plotting_units = kwargs.pop('plot_units', None)
+        u, v = self._plotting_units(u, v, plotting_units)
+
+        defaults = {'pivot': 'tail', 'scale': 20, 'scale_units': 'inches', 'width': 0.002}
+        defaults.update(kwargs)
+
+        # Remove old arrows
+        if self.arrows:
+            self.arrows.remove()
+
+        # Handle transforming our center points. CartoPy doesn't like 1D arrows
+        # TODO: This can be removed for cartopy > 0.14.3
+        if hasattr(self.ax, 'projection') and 'transform' in kwargs:
+            trans = kwargs['transform']
+            try:
+                kwargs['transform'] = trans._as_mpl_transform(self.ax)
+            except AttributeError:
+                pass
+            u, v = self.ax.projection.transform_vectors(trans, self.x, self.y, u, v)
+
+            # Since we've re-implemented CartoPy's arrows, we need to skip calling it here
+            self.arrows = matplotlib.axes.Axes.quiver(self.ax, self.x, self.y, u, v,
+                                                      **defaults)
+        else:
+            self.arrows = self.ax.quiver(self.x, self.y, u, v, **defaults)
```

Comment:
Would be nice to find out if we can nuke this block, given that CartoPy 0.17.0 is a year old already.

Reply:
I think the transform in the barbs or quiver calls only work to transform where the vector is plotted, not transforming the actual vector. See the example of 500-hPa Absolute Vorticity from the gallery - https://unidata.github.io/python-gallery/examples/500hPa_Absolute_Vorticity_winds.html . In order to get the correct winds, need to both transform the vector, then plot them in the correct location with a transform in barbs.

Reply:
So I guess it’s possible that our hack to work-around support for 1D arrays was always wrong?